### ISSUE TITLE
After a Network Connection is established how server to determine the connection is active?

### ISSUE THREAD

**Issue by ioprotocol** [vert-x3/vertx-mqtt-server#30]
Hi, if  a Network Connection is established,but the client do not send a CONNECT packet , according to MQTT 3.1.1 specification, the server should close the Network connection after a reasonable amount of time, In this case, How to solve this problem?
I think should edit in here,
```java
@Override
  public MqttServer listen(int port, String host, Handler<AsyncResult<MqttServer>> listenHandler) {
    Handler<MqttEndpoint> h1 = endpointHandler;
    Handler<Throwable> h2 = exceptionHandler;
    server.connectHandler(so -> {
      NetSocketInternal soi = (NetSocketInternal) so;
      ChannelPipeline pipeline = soi.channelHandlerContext().pipeline();
      // here add a IdleStateHandler to close the inactive connection,and remove theIdleStateHandler  when CONNECT packet arrive.
      pipeline.addBefore("handler", "idle", new IdleStateHandler(0, 0, timeout));
      pipeline.addBefore("handler", "mqttEncoder", MqttEncoder.INSTANCE);
      if (this.options.getMaxMessageSize() > 0) {
        pipeline.addBefore("handler", "mqttDecoder", new MqttDecoder(this.options.getMaxMessageSize()));
      } else {
        // max message size not set, so the default from Netty MQTT codec is used
        pipeline.addBefore("handler", "mqttDecoder", new MqttDecoder());
      }

      MqttConnection conn = new MqttConnection(soi, options);

      soi.messageHandler(msg -> {
        conn.handleMessage(safeObject(msg, soi.channelHandlerContext().alloc()));
      });

      conn.init(h1, h2);

    });
    server.listen(port, host, ar -> listenHandler.handle(ar.map(this)));
    return this;
  }
```
what do yo think about this ?

**Comment by ppatierno** [comment]
With the latest commit, the idle handler was added for the PINGREQ/PINGRESP but that's true that it's injected only after the CONNECT packet. Good point ! I'll take a look at it ;)

**Comment by ioprotocol** [comment]
Another question is in MqttConnection handleMessage, according to MQTT 3.1.1 specification,the server received first packet must be CONNECT packet, if not the Server should close the network connection and a network connection only permit to send CONNECT packet once, so in handleMessage I think should add a checkpoint like this:
```java
synchronized void handleMessage(Object msg) {

    // handling directly native Netty MQTT messages because we don't need to
    // expose them at higher level (so no need for polyglotization)
    if (msg instanceof io.netty.handler.codec.mqtt.MqttMessage) {

      io.netty.handler.codec.mqtt.MqttMessage mqttMessage = (io.netty.handler.codec.mqtt.MqttMessage) msg;

      DecoderResult result = mqttMessage.decoderResult();
      if (result.isFailure()) {
        chctx.pipeline().fireExceptionCaught(result.cause());
        return;
      }
      if (!result.isFinished()) {
        chctx.pipeline().fireExceptionCaught(new Exception("Unfinished message"));
        return;
      }
    
     // here can like this
    if(endpoint == null && mqttMessage.fixedHeader().messageType() != CONNECT) {
          // close the network connection
    }
    if(endpoint != null && mqttMessage.fixedHeader().messageType() == CONNECT) {
         // if go here, indicate that the client send CONNECT packet twice, which is also need to close the network connection.
    }
    switch (mqttMessage.fixedHeader().messageType()) {

        case CONNECT:
          handleConnect((MqttConnectMessage) msg);
          break;

        case PUBACK:

          io.netty.handler.codec.mqtt.MqttPubAckMessage mqttPubackMessage = (io.netty.handler.codec.mqtt.MqttPubAckMessage) mqttMessage;
          this.handlePuback(mqttPubackMessage.variableHeader().messageId());
          break;

        case PUBREC:
```
Thanks for your reply!

**Comment by ppatierno** [comment]
It's fixed on behalf of 35b43bfe65b22c21ae2ffcaea053f96c7dc94a3f
